app/helpers/rules_helper.rb

```diff
@@ -20,4 +20,12 @@ def flatten_payload_values(payload, prefix = nil)
       { prefix => payload }
     end
   end
+
+  def available_handlers
+    current_shop_handlers = shop.rules.map { |r| r.handlers.map { |h| h.service_name } }.flatten.uniq
```

Comment:
Maybe something like?

```ruby
current_shop_handlers = shop.rules.joins(:handlers).select('handlers.service_name').distinct
(Handler::HANDLERS.map(&:to_s) + current_shop_handlers).uniq
```

Reply:
I would need to call `.map(&:constantize)` and would like to avoid id.

Reply:
Ah, you need access to their labels, I see. I'd still favor `shop.rules.joins(:handlers).select('handlers.service_name').distinct
` over the double map, but otherwise I'm good with the following loop.